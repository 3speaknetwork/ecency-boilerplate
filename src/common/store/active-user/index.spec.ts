import reducer, { initialState, loginAct, logoutAct, updateAct } from "./index";

jest.mock("../../util/local-storage", () => ({
  get: (k: string) => {
    if (k === "active_user") {
      return "foo";
    }

    if (k.startsWith("user_")) {
      return "....";
    }

    return null;
  },
}));

let state = initialState;

it("1- default state", () => {
  expect(state).toMatchSnapshot();
});

it("2- loginAct", () => {
  state = reducer(state, loginAct());
  expect(state).toMatchSnapshot();
});

it("3- updateAct", () => {
  state = reducer(state, updateAct({ name: "foo", post_count: 10, __loaded: true }));
  expect(state).toMatchSnapshot();
});
