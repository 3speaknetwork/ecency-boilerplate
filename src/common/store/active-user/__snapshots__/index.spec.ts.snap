// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `null`;

exports[`2- reloadAct 1`] = `
Object {
  "data": Object {
    "name": "foo",
  },
  "username": "foo",
}
`;

exports[`3- updateAct 1`] = `
Object {
  "data": Object {
    "__loaded": true,
    "name": "foo",
    "post_count": 10,
  },
  "username": "foo",
}
`;
